mitiq/calibration/calibrator.py

```diff
@@ -92,208 +94,118 @@ def get_cost(self) -> Dict[str, int]:
             "ideal_executions": ideal,
         }
 
-    def run_circuits(self) -> List[Dict[str, Any]]:
+    def run_circuits(self) -> List[Result]:
         """Runs all the circuits required for calibration.
 
         Returns:
             A collection of experimental results along with a summary of each
             :class:`BenchmarkProblem` that was run.
         """
-        expvals = []
+        results = []
+
+        strategies = self.settings.make_strategies()
         for problem in self.circuits:
-            circuit, distribution = problem["circuit", "ideal_distribution"]
+            circuit = problem.circuit
             circuit.append(cirq.measure(circuit.all_qubits()))
 
-            expval_executor, bitstring_to_measure = convert_to_expval_executor(
-                self.executor, distribution
+            bitstring_to_measure = problem.most_likely_bitstring()
+            expval_executor = convert_to_expval_executor(
+                self.executor, bitstring_to_measure
             )
+
             noisy_value = expval_executor.evaluate(circuit)[0]
-            ideal_value = distribution[bitstring_to_measure]
-            noisy_error = ideal_value - noisy_value
-
-            mitigated_values: Dict[str, Dict[str, Any]] = {
-                technique.name: {"results": [], "improvement_factor": None}
-                for technique in self.settings.techniques
-            }
-            for strategy in self.settings.make_strategies():
+
+            for strategy in strategies:
                 mitigated_value = strategy.mitigation_function(
                     circuit, expval_executor
                 )
-                improvement_factor = abs(
-                    noisy_error / (ideal_value - mitigated_value)
-                )
 
                 result = {
-                    "mitigated_value": mitigated_value,
-                    "improvement_factor": improvement_factor,
-                    "strategy": strategy,
-                    **strategy.as_dict(),
-                }
-                technique = strategy.technique.name
-                mitigated_values[technique]["results"].append(result)
-
-            circuit_info = problem.problem_summary_dict()
-            expvals.append(
-                {
-                    "circuit_info": circuit_info,
-                    "noisy_value": noisy_value,
-                    "ideal_value": distribution[bitstring_to_measure],
-                    "mitigated_values": mitigated_values,
+                    "mitigated": mitigated_value,
+                    "noisy": noisy_value,
+                    "ideal": problem.largest_probability(),
```

Comment:
This information is already in `problem` is it necessary to copy it also in `results`?

Reply:
This is true, but it's often convenient to have all of mitigated/noisy/ideal values in the same place. When this is not the case, it's much less convenient to compute errors due to the fact that we are not storing entire `Strategy` and `BenchmarkProblem` instances, but rather unique IDs. Thus we would have to grab the `BenchmarkProblem` ID, get the actual instance, get the ideal value, and execute it to obtain the noisy value.

While I dislike duplicating data, I think this makes the code more readable, and maintainable.